packages/next/build/webpack/plugins/next-esm-plugin.ts

```diff
@@ -0,0 +1,258 @@
+/**
```

Comment:
This plugin appears to be very fragile and tightly integrated into webpack's internals.

It seems to be simulating a third compiler, e.g. [re-running all provided plugins](https://github.com/zeit/next.js/pull/7704/files#diff-c8386495c7607bdb08870b52e32c7e87R199) and [re-computing the chunk graph](https://github.com/zeit/next.js/pull/7704/files#diff-c8386495c7607bdb08870b52e32c7e87R230).

If my understanding is correct, it's also re-babeling the individual files rather than chunks as a whole (effectively rebundling instead of post-processing emitted chunks).

The only way I feel we could justify this plugin is if it post-processed already computed chunks, however, that comes with its own problems like skipping the webpack plugin flow for user customizations (e.g. service worker, etc).

What's the real value we're deriving from this? It seems to me we're rather instantiate a third webpack instance. This would greatly simplify the implementation and delete the need for this plugin.

----

If the upside to this plugin is to save "rework" within `node_modules`, I'd argue that point is moot. AFAIK the ecosystem is wanting to move to a place where all npm modules are published in ES.Latest, meaning `node_modules` will need to be compiled (and then re-compiled for the modern bundle).

Please let me know if I've misunderstood something!